src/components/QueryPage.tsx

```diff
@@ -159,6 +155,10 @@ const QueryPage = ({
                 setHasResults(true);
                 setIsEdit(false);
               }}
+              setHasResults={() => {
+                setHasResults(true);
+                onRefresh();
```

Comment:
I'd be interested in discussing whether it's worthwhile to extend this via something like

- `useContext`
- `window.roamjs.extension.queryBuilder`
- or another way

so that we could use `addCommand` instead of the `EventListener` in QueryEditor